client/asset/btc/btc.go

```diff
@@ -1945,21 +1928,28 @@ func (btc *ExchangeWallet) send(address string, val uint64, feeRate uint64, subt
 	return nil, 0, 0, fmt.Errorf("failed to locate transaction vout")
 }
 
-// Confirmations gets the number of confirmations for the specified coin ID.
-// The coin must be known to the wallet, but need not be unspent.
-func (btc *ExchangeWallet) Confirmations(_ context.Context, id dex.Bytes) (uint32, error) {
-	txHash, _, err := decodeCoinID(id)
+// Confirmations gets the number of confirmations for the specified coin ID by
+// first checking for a unspent output, and if not found, searching indexed
+// wallet transactions.
+func (btc *ExchangeWallet) Confirmations(_ context.Context, id dex.Bytes) (confs uint32, spent bool, err error) {
+	txHash, vout, err := decodeCoinID(id)
 	if err != nil {
-		return 0, err
+		return 0, false, err
 	}
+	// Check for an unspent output.
+	txOut, err := btc.node.GetTxOut(txHash, vout, true)
+	if err == nil && txOut != nil {
+		return uint32(txOut.Confirmations), false, nil
+	}
+	// Check wallet transactions.
 	tx, err := btc.wallet.GetTransaction(txHash.String())
 	if err != nil {
 		if isTxNotFoundErr(err) {
-			return 0, asset.CoinNotFoundError
+			return 0, true, asset.CoinNotFoundError
 		}
-		return 0, err
+		return 0, true, err
```

Comment:
If gettransaction can't find it, I don't think we should return `spent=true`

Reply:
With an non-nil error, it the caller should probably be ignoring the value anyway, so I'll defer to the zero value in that case. We can't say that it's a spent output anyway, though we can say that it's not an unspent output.